ansible/roles/iroha-docker/files/iroha_utils.py

```diff
@@ -105,24 +106,61 @@ def gen_keys(number=1):
         result.append({'pub': pub.decode('ascii'), 'priv': priv.decode('ascii')})
     return result
 
+
 def ansible_vault_encrypt(key, text):
     vault = VaultLib([(DEFAULT_VAULT_ID_MATCH, VaultSecret(key.rstrip('\n').encode('utf-8')))])
     return vault.encrypt(text.rstrip('\n').encode('utf-8')).decode('utf-8')
 
-if ('DEBUG' in os.environ) and  os.environ['DEBUG'] == 'TRUE':
+
+def filter_old_peers(block: dict):
+    transactions = block['block_v1']['payload']['transactions']
+    for i in range(len(transactions)):
+        tx_commands = transactions[i]['payload']['reducedPayload']['commands']
+        clean_commands = filter(lambda command_list: 'addPeer' not in command_list.keys(), tx_commands)
+        transactions[i]['payload']['reducedPayload']['commands'] = list(clean_commands)
+    return block
+
+
+def add_new_peers(block: dict, new_peers: list):
+    initial_tx_commands = block['block_v1']['payload']['transactions'][0]['payload']['reducedPayload']['commands']
```

Comment:
You will change the original value of "new_block" variable in global area if you don't make a deep copy of "block" variable here. It breaks nothing for this situation, but you lose original value for the value of "new_block" variable.

Reply:
I did it on purpose and added the extra variables to make it more understandable. I will make it clearer,